fili-core/src/test/groovy/com/yahoo/bard/webservice/data/dimension/impl/SearchProviderSpec.groovy

```diff
@@ -526,6 +529,34 @@ abstract class SearchProviderSpec<T extends SearchProvider> extends Specificatio
         resultsPage.numResults == 3
     }
 
+    def "An empty string filter value should return all results"() {
+        setup: "Given a filter that will match every row"
+        /* Expected rows, not necessarily in this order:
+                name: "owl", description: "this is an owl"
+                name: "hawk", description: "this is a raptor"
+                name: "eagle", description: "this is a raptor"
+                name: "kumquat", description: "this is not an animal"
+        */
+        Set<ApiFilter> filters = [
+                buildFilter("animal|desc-startswith[this]"),
+                buildFilter('animal|desc-contains[""]')
+        ]
+        Set<String> expectedOutcomes = ["owl", "hawk", "eagle", "kumquat"]
+        and: "We get the second page, where each page has two rows"
+        PaginationParameters parameters = new PaginationParameters(4, 1)
+
+        when: "We query the search provider"
+        Pagination<DimensionRow> resultsPage = searchProvider.findFilteredDimensionRowsPaged(filters, parameters)
+
+        then: "We get only one page of results"
+        resultsPage.getPageOfData().size() == 4
+
+        and: "We get only {'owl', 'hawk', 'eagle', 'kumquat'}"
+        resultsPage.numResults == 4
+        Set<String> actualOutcomes = resultsPage.pageOfData.stream().map { it.getKeyValue() }.collect(Collectors.toSet())
+        actualOutcomes.equals(expectedOutcomes)
+    }
```

Comment:
To make this test more idiomatic for groovy and spock, lines 564-5 could be rewritten as follows:
expectedOutcomes == resultsPage.pageOfData.collect { it.getKeyValue() } as Set

This uses the simpler groovy collect without going through java streams of collectors library and the coercion operator to control the collection type.
The use of '==' is (sort of) equivalent to .equals but leverages the power assert capability for more useful debugging info.